jobs/commons/src/services/read-model.service.ts

```diff
@@ -1,13 +1,21 @@
-import { Db, MongoClient, MongoClientOptions, ReadPreferenceMode } from 'mongodb'
-import { ReadModelConfig } from '../index.js'
+import { Collection, Db, MongoClient, MongoClientOptions, ReadPreferenceMode } from 'mongodb'
+import { ATTRIBUTES_COLLECTION_NAME, Attribute, ESERVICES_COLLECTION_NAME, EService, PURPOSES_COLLECTION_NAME, Purpose, ReadModelConfig, TENANTS_COLLECTION_NAME, Tenant } from '../index.js'
 
 export class ReadModelClient {
   private mongodbClient: MongoClient
-  private database: Db
+
+  eservices: Collection<{ data: EService }>
+  tenants: Collection<{ data: Tenant }>
+  purposes: Collection<{ data: Purpose }>
+  attributes: Collection<{ data: Attribute }>
 
   private constructor(mongodbClient: MongoClient, db: Db) {
     this.mongodbClient = mongodbClient
-    this.database = db
+
+    this.attributes = db.collection<{ data: Attribute }>(ATTRIBUTES_COLLECTION_NAME)
+    this.eservices = db.collection<{ data: EService }>(ESERVICES_COLLECTION_NAME)
+    this.tenants = db.collection<{ data: Tenant }>(TENANTS_COLLECTION_NAME)
+    this.purposes = db.collection<{ data: Purpose }>(PURPOSES_COLLECTION_NAME)
```

Comment:
Why not a getter function? 

```typescript
get eservices(): Collection<{ data: EService }> {
    return this.db.collection(ESERVICES_COLLECTION_NAME) // { data: EService } should be inferred automatically from the return type
}

```

Reply:
```typescript
// private mongodbClient: MongoClient
// private database: Db 
// ^^^^^^^^^
// No need to specify the parameters here, this a typescript's syntactic sugar (https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties) 
private constructor(private mongodbClient: MongoClient, private db: Db) {}

get eservices(): Collection<{data: EService}> {
  return this.db.collection(ESERVICES_COLLECTION_NAME)
}

// ...

```

Reply:
> Why not a getter function?

It's the instinct for optimizations, even if minor 😂 
Each time `db.collection` is invoked a new object is created, performing several checks, and creating other objects. In our case the collections are fixed so we do not need them to run on each query execution.
It has a low impact on jobs but every little piece helps on long running processes with lots of requests 😄 

But if you see any drawback let's talk about it

Reply:
No drawbacks! Thanks for the explanation 😁